backend/app/services/cassandra_service.py

```
@@ -0,0 +1,724 @@
+            import json
+
+            loop = asyncio.get_event_loop()
+            query_hash = hashlib.md5(search_query.lower().encode()).hexdigest()
+
+            select_query = f"""
+                SELECT results, expires_at
+                FROM {self.keyspace}.user_search_results
+                WHERE search_query_hash = ? AND client_account_id = ? AND user_id = ?
+                AND expires_at > ?
+                LIMIT 1
+            """
```

Ruff (0.14.10)

Possible SQL injection vector through string-based query construction

(S608)